docs/plugins/data/query-editor-enhancements.md

```diff
@@ -46,14 +46,14 @@ export interface QueryEditorExtensionConfig {
    * @param dependencies - The dependencies required for the extension.
    * @returns The query editor extension component.
    */
-  getComponent?: (dependencies: QueryEditorExtensionDependencies) => React.ReactElement | null;
+  getComponent?: (dependencies: QueryEditorExtensionDependencies) => Promise<React.ReactElement> | React.ReactElement | null;
```

Comment:
one reason I don't want to have async components is that it does not explicitly require consumers to define loading states (like fallback components with Suspense). I think they are important for user experience, and consumer should decide if they should be null or should be something else